extensions/wikia/CrosslinkModule/CrosslinkModuleHelper.class.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/**
+ * Class CrosslinkModuleHelper
+ */
+class CrosslinkModuleHelper extends WikiaModel {
+
+	const CACHE_TTL = 86400;
+
+	/**
+	 * Show the module only on article pages (for Anon users only)
+	 * @return boolean
+	 */
+	public function canShowModule() {
+		if ( $this->wg->User->isLoggedIn() || !$this->app->checkSkin( 'oasis' ) ) {
+			return false;
+		}
+
+		$title = $this->wg->Title;
+		if ( $title instanceof Title && $title->getNamespace() == NS_MAIN ) {
+			return in_array( $title->getArticleID(), $this->getValidPages() );
+		}
+
+		return false;
+	}
+
+	/**
+	 * Get list of pages that the module is enabled
+	 * @return array
+	 */
+	protected function getValidPages() {
+		$cacheKey = wfMemcKey( 'crosslink_module', 'valid_pages' );
+		$db = wfGetDB( DB_SLAVE, [], 'specials' );
+		$pages = (new WikiaSQL())->cache( static::CACHE_TTL, $cacheKey, true )
+			->SELECT( 'source_page' )
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "source_page" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)